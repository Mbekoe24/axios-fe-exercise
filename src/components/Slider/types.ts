import { SlideProps } from "components/Slide/types";

export interface ApiResponse extends SlideProps{
  id: string
}

export interface SliderProps {
  data: SlideProps[],
}