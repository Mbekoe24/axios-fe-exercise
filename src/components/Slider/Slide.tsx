import React from "react";
import { SlideProps } from "../../../types";

export const Slide = (props: SlideProps) => {
  return (

    ////// testing if data populates
    <div className="slides">
      <div id="slide-1">
        <a href={props.permaLink}>
          <img src={props.primaryImage} alt="" />
        </a>
        <h1>{props.headline}</h1>
        <span>
          {props.publishedDate} - {props.sectionLabel}
        </span>
      </div>
      <div id="slide-2">2</div>
      <div id="slide-3">3</div>
      <div id="slide-4">4</div>
      <div id="slide-5">5</div>
    </div>
  );
};
