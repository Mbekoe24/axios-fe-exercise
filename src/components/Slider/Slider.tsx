import React from "react";
import styles from "./Slider.module.scss";
import { SliderProps, ApiResponse } from "./types";
import { Slide } from "../Slide";

const Slider = ({ data }: SliderProps) => {
  return (
    // loop through to create a link each
    // <a href="#slide-1">1</a>
    // <a href="#slide-2">2</a>
    // <a href="#slide-3">3</a>
    // <a href="#slide-4">4</a>
    // <a href="#slide-5">5</a>
    <div className={styles["slider"]}>
      {data && (
        <>
          {data?.map((item: ApiResponse) => {
            console.log(item.headline);
            return (
              <Slide
                key={item.id}
                headline={item.headline}
                displayName={item.displayName}
                sectionLabel={item.sectionLabel}
                publishedDate={item.publishedDate}
                primaryImage={item.primaryImage}
                permaLink={item.permaLink}
              />
            );
          })}
        </>
      )}
    </div>
  );
};

export default Slider;
