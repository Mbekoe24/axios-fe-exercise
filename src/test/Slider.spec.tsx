import { render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import Slider from "../components/Slider/Slider";
import * as React from "react";

describe("Slider on page", () => {
  it("it matches snapshot", () => {
    const { container } = render(
      <Slider
        data={[
          {
            htmlId: "",
            headline: "",
            displayName: "",
            topicName: "",
            altText: "",
            publishedDate: "",
            primaryImage: "",
            permaLink: "",
          },
        ]}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("it shows Vist axios button on page snapshot", async () => {
    <Slider
      data={[
        {
          htmlId: "",
          headline: "",
          displayName: "",
          topicName: "",
          altText: "",
          publishedDate: "",
          primaryImage: "",
          permaLink: "",
        },
      ]}
    />;

    const button = await screen.findByText("Visit Axios.com");
    expect(button).toBeInTheDocument();
    user.click(button);
    expect(button).toBeEnabled();
  });
});
