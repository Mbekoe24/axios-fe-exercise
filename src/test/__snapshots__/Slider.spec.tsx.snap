// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider it matches snapshot 1`] = `
Array [
  <h1>
    More from Axios.com
  </h1>,
  <div
    className=" news-slider"
  >
    <div
      className="slides"
    >
      <div
        id="slide-1"
      >
        <a
          href="{item.permaLink}"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="{item.altText}"
            src="{item.primaryImage}"
          />
        </a>
        <div
          className="info-box"
        >
          <p>
            {item.headline}
          </p>
          <span>
            <time>
              Invalid Date
            </time>
             - 
            {item.topicName}
            
          </span>
        </div>
      </div>
    </div>
    <div
      className="pagination-container"
    >
      <a
        className="pagination"
        href="#slide-1"
      >
        1
      </a>
    </div>
  </div>,
  <button
    className="axios-button"
    onClick={[Function]}
  >
    Visit Axios.com
  </button>,
]
`;
